backend/app/services/adapters/executor_job.py

```
@@ -131,18 +131,58 @@ def cleanup_stale_executors(self, db: Session) -> None:
+
+                    if task_type == "code":
+                        # Try to archive workspace before deletion
+                        try:
+                            self._archive_workspace_sync(
+                                db=db,
+                                subtask=subtask,
+                                task=task,
+                                executor_name=name,
+                                executor_namespace=ns,
+                            )
+                        except Exception as archive_error:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)